streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -39,7 +40,7 @@
                                final KeySchema keySchema) {
         this.name = name;
         this.keySchema = keySchema;
-        this.segments = new Segments(name, retention, numSegments);
+        this.segments = new Segments(name, retention, Segments.segmentInterval(retention, numSegments));
```

Comment:
Why this change? The existing code computes segmentInterval internally already. Seems to be better contained, as we pass in `rentention` time anyway?

Reply:
Oof, while doing this I discovered a bug in which the SegmentedCache takes a segmentInterval argument, but the WindowBytesStoreSupplier provides instead the *number of segments* to it.

Here's the call passing #segments: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/state/internals/WindowStoreBuilder.java#L55

Here's the receiving end of the call: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java#L59

The types are different, but it's int -> long, so Java automatically up-casts it. :(

There's no way to fix that bug without altering the public interface. I guess the right thing to do would be to deprecate https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/state/WindowBytesStoreSupplier.java#L37 and add a `segmentInterval` method instead. 

It's ironic because there is exactly one call to that method in the entire codebase, and it uses it wrong :/

What do you think I should do about this?

Reply:
I considered deviously just "correcting" the JavaDoc to state that `segments` is the segment interval, but I'm afraid that external code might either be implementing this interface or using it (for a custom store impl)...

Reply:
I've added a fix for this to this issue (https://github.com/apache/kafka/pull/5253/commits/1d0359614b5444230cd99e7c0e02a0978da81435), but I should probably file a jira just for this issue and write a small KIP proposing the interface change and then create a separate PR just containing https://github.com/apache/kafka/pull/5253/commits/1d0359614b5444230cd99e7c0e02a0978da81435.

WDYT?

Reply:
Squinting at the `WindowBytesStoreSupplier` now.. with the current API we are actually enforcing any customized Window bytes store in DSL's `Materialized` to follow the same segmented pattern as `RocksDbWindowBytesStoreSupplier`, right? Ideally a customized window store supplier only need to expose the window size, and maybe retention period. Of course it is a general issue that may need a separate discussion...

For now I think adding retention interval and deprecating num.segments in `WindowBytesStoreSupplier` is okay, but it would require a KIP.